### PR TITLE
task/mosaic-restrictions

### PR DESCRIPTION
**Pull request by gimer**

### CODE REVIEW

File: catparser/CatsParser.py [hunk 1]
```diff
@@ -53,7 +53,10 @@ def _process_line(self, line):
 
         active_factories = self.type_parser_factories if not self.active_parser else self.active_parser.factories()
 
-        factory = next(factory for factory in active_factories if factory.is_match(line_stripped))
+        try:
+            factory = next(factory for factory in active_factories if factory.is_match(line_stripped))
+        except StopIteration:
+            raise CatsParseException('none of the parsers matched the line "{0}"'.format(line_stripped))
```
Comment (Jaguar0625) [inline catparser/CatsParser.py]: newline after raise

File: generators/cpp_builder/CppGenerator.py [hunk 2]
```diff
@@ -95,7 +95,7 @@ def _load_hints(filenames):
         all_hints = {}
         for filename in filenames:
             with open('generators/cpp_builder/hints/{0}.yaml'.format(filename)) as input_file:
-                hints = yaml.load(input_file)
+                hints = yaml.load(input_file, Loader=yaml.SafeLoader)
```
Comment (Jaguar0625) [inline generators/cpp_builder/CppGenerator.py]: why this change?

File: schemas/restriction_mosaic/mosaic_address_restriction.cats [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import "transaction.cats"
+import "restriction_mosaic/mosaic_restriction_types.cats"
```
Comment (Jaguar0625) [inline schemas/restriction_mosaic/mosaic_address_restriction.cats]: mosaic_restriction_types not needed

File: schemas/restriction_mosaic/mosaic_global_restriction.cats [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import "transaction.cats"
+import "restriction_mosaic/mosaic_restriction_types.cats"
```
Comment (Jaguar0625) [inline schemas/restriction_mosaic/mosaic_global_restriction.cats]: swap imports

File: schemas/restriction_mosaic/mosaic_address_restriction.cats [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+# binary layout for a mosaic address restriction transaction
+struct MosaicAddressRestrictionTransactionBody
+	# identifier of the mosaic to which the restriction applies
+	mosaicId = UnresolvedMosaicId
+
+	# restriction key
+	restrictionKey = uint64
+
+	# address being restricted
+	targetAdderss = UnresolvedAddress
```
Comment (Jaguar0625) [inline schemas/restriction_mosaic/mosaic_address_restriction.cats]: Address

File: schemas/restriction_mosaic/mosaic_address_restriction.cats [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+	# previous restriction value
+	previousRestrictionValue = uint64
+
+	# new restriction value
+	newRestrictionValue = uint64
+
+# binary layout for a non-embedded mosaic global restriction transaction
```
Comment (Jaguar0625) [inline schemas/restriction_mosaic/mosaic_address_restriction.cats]: global -> address (x2)

File: schemas/restriction_mosaic/mosaic_restriction_types.cats [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+	# allow if equal
+	eq = 1
+
+	# allow if not equal
+	ne = 2
+
+	# allow if less than
+	lt = 3
+
+	# allow if less than or equal.
```
Comment (Jaguar0625) [inline schemas/restriction_mosaic/mosaic_restriction_types.cats]: no periods

**Comment by Jaguar0625** [comment]
please fix the new tx entity types

File: generators/cpp_builder/CppGenerator.py (hunk 2 above)
Comment (gimer) [reply generators/cpp_builder/CppGenerator.py]: python was complaining, that yaml.load without specifying loader is deprecated
